XTM722/Planetze
# feat: Add IntroActivity and user-first navigation flow

## Description

This Pull Request adds the `IntroActivity` for first-time users, adjusts the navigation logic in `MainActivity`, and includes UI updates with improved drawable resources. The onboarding experience has been enhanced to ensure smooth navigation for both new and returning users.

### Key Changes:
- Implemented `IntroActivity` to onboard first-time users.
- Updated `MainActivity` to determine navigation flow based on user status.
- Designed the layout (`activity_intro.xml`) with clear onboarding messages and actions.
- Added new drawable assets for icons and backgrounds.
- Updated `strings.xml` with relevant onboarding text.

---
## Checklist

- [x] Code builds successfully without errors.
- [x] No crashes or runtime errors during manual testing.
- [x] All navigation flows work as intended for first-time and returning users.
- [x] Strings are externalized in `strings.xml` (no hardcoded values).
---
## Testing Instructions

1. Launch the app in an emulator or physical device.
2. Log in with a **new account** to verify that the app navigates to `IntroActivity`.
3. Complete onboarding and verify navigation to `QuestionActivity`.
4. Log in with an **existing account** to ensure the app skips `IntroActivity` and navigates directly to `DashboardActivity`.
5. Verify that all UI elements, icons, and text are correctly displayed.

### Testing Method
- **Manual Testing**: The functionality was verified by navigating through the app on the Android Emulator and ensuring expected behaviors.

---

## Additional Notes

- This PR focuses on the onboarding flow. Future iterations may include animations for a better user experience.
- Manual testing was performed on Android Emulator
- Feedback on improving text or layout in `IntroActivity` is welcome.
